test/runner/browserstack/api.js

```diff
@@ -148,17 +158,54 @@ export async function filterBrowsers( filter ) {
 	const filterOsVersion = ( filter.os_version ?? "" ).toLowerCase();
 	const filterDevice = ( filter.device ?? "" ).toLowerCase();
 
-	return browsers.filter( ( browser ) => {
+	const filteredWithoutVersion = browsers.filter( ( browser ) => {
 		return (
 			( !filterBrowser || filterBrowser === browser.browser.toLowerCase() ) &&
-			( !filterVersion ||
-				matchVersion( browser.browser_version, filterVersion ) ) &&
 			( !filterOs || filterOs === browser.os.toLowerCase() ) &&
-			( !filterOsVersion ||
-				filterOsVersion === browser.os_version.toLowerCase() ) &&
+			( !filterOsVersion || matchVersion( browser.os_version, filterOsVersion ) ) &&
 			( !filterDevice || filterDevice === ( browser.device || "" ).toLowerCase() )
 		);
 	} );
+
+	if ( !filterVersion ) {
+		return filteredWithoutVersion;
+	}
+
+	if ( filterVersion.startsWith( "latest" ) ) {
+		const groupedByName = filteredWithoutVersion
```

Comment:
Is it worth to create such a structure when we always need the browser name specified? It doesn't make sense to support "latest" without the browser name so we could just filter by name, right?

Reply:
I'm not sure what you're asking. The list API will return all matching browsers even when there's no name.

Reply:
Right, but you're creating an object mapping browser names to an array of matching browser object. Since we know the browser name we want, why not just filter the browser we want without any grouping instead of transforming everything?

Reply:
Actually, you can't assume we know the browser name. And the point is to ensure there is only one per browser name returned.